PSScriptAnalyzer 1.22.0 (#200)
MicrosoftDocs/PowerShell-Docs-Modules: reference/docs-conceptual/PSScriptAnalyzer/Rules/AvoidExclaimOperator.md

---
description: Avoid exclaim operator
ms.custom: PSSA v1.22.0
ms.date: 06/14/2023
ms.topic: reference
title: AvoidExclaimOperator
---
# AvoidExclaimOperator
**Severity Level: Warning**

## Description

> The negation operator `!` should not be used for readability purposes. Use `-not` instead.
> 
> **Note**: This rule is not enabled by default. The user needs to enable it through settings.

Reviewer: ```suggestion
Avoid using the negation operator (`!`). Use `-not` for improved readability.

> [!NOTE]
> This rule is not enabled by default. The user needs to enable it through settings.
```
